### PR TITLE
Fix type errors found in sentry

### PR DESCRIPTION
**Pull request by nh758**
Fix sentry type errors:
- [APPBUILDER-WEB-1M0](https://appdev-designs.sentry.io/issues/4904347622/)
- [APPBUILDER-WEB-25E](https://appdev-designs.sentry.io/issues/5343008152/)
- [APPBUILDER-WEB-2ED](https://appdev-designs.sentry.io/issues/6016419585/)
- [APPBUILDER-WEB-2CK](https://appdev-designs.sentry.io/issues/5834834957/)

## Release Notes

- fix type errors

### CODE REVIEW

**Review by guyyoo — APPROVED** [review]
LGTM!

File: AppBuilder/platform/FilterComplex.js [hunk 1]
```diff
@@ -494,6 +494,7 @@ module.exports = class FilterComplex extends FilterComplexCore {
          this._fnBaseGetValue = $filterView.prototype.GetValue;
       $filterView.prototype.GetValue = function () {
          const rule = _this._fnBaseGetValue.call(this);
+         if (!rule) return;
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/FilterComplex.js:497]: is this an expected case?

File: AppBuilder/platform/views/viewComponent/ABViewDataviewComponent.js [hunk 2]
```diff
@@ -82,6 +82,7 @@ module.exports = class ABViewDataviewComponent extends ABViewComponent {
 
    resize(base_element) {
       const $dataview = $$(this.ids.dataview);
+      if (!$dataview) return;
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/views/viewComponent/ABViewDataviewComponent.js:85]: is this an expected case?

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
I don't know the answer to this, so I'm just asking here.

What you're doing is perfectly fine for preventing the errors due to referencing `null` or `undefined` values.

However, in these cases, is a `null` or `undefined` value a reasonable result? Or is that an anomaly and should there be some notification about the result?

If they are reasonable results, then everything is good.  Let's go!

otherwise, let's put in some warning so we might be able to identify some strange edge cases when they are happening.

**Pull request merged**